### DISCUSSION TITLE
How to submit patches or pull requests?

### DISCUSSION THREAD

**Discussion by maphew** [pcmacdon/jsish#45] [Q&A] [answered]
@pcmacdon what is your preferred route for receiving patch suggestions and/or pull requests?

I surmise the primary code management is via Fossil at Jsish.org and this github repo is a oneway export mirror? I'm not sure what would be the smoothest contribution workflow. There's no hurry; for the moment I'm thinking mostly of doc improvement. I may have spotted one small error in a .jsi script, but that may well be just my ignorance also.

**Comment by pcmacdon** [comment, marked as answer]
Yes, jsish is just a one-way export mirror, because Fossil is used for development.  So the easist way to submit simple changes is with:

   fossil diff > diff.pat

More complex changes may want to use "fossil patch" which was new in 2.16 released in July.  I just discovered this but it looks like this:

   fossil patch create diff.fospat

**Comment by pcmacdon** [comment]
Note that I have just myself upgraded to fossil 2.16, which I've been waiting or since I reported [this fossil open bug](https://fossil-scm.org/forum/forumpost/37b59b768371ad31372c77d251b721e0256512c8af277d246bb9a9ee9e673154) 5 months ago.

The fix plus the new "patch" subcommand has implications for PDQ, particularly for syncing.  Stay tuned.
